### PR TITLE
Temp

### PR DESCRIPTION
**Pull request by chamil321**
stat integration

### CODE REVIEW

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml [hunk 1]
```diff
@@ -31,5 +32,106 @@
...
+        <private.package>org.wso2.carbon.transport.http.netty.statistics.internal</private.package>
+        <export.package>
+            !org.wso2.carbon.transport.http.netty.statistics.internal,
+            org.wso2.carbon.transport.http.netty.statistics,
+        </export.package>
+        <import.package>
+            org.wso2.carbon.transport.http.netty.statistics,
+            org.osgi.framework.*;version="${osgi.framework.import.version.range}",
+            org.osgi.util.tracker;version="${osgi.service.tracker.import.version.range}",
+            <!--org.wso2.carbon.kernel.transports.*;version="${carbon.kernel.package.import.version.range}",-->
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml]: I guess we can remove these commented lines.

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml [hunk 2]
```diff
@@ -31,5 +32,106 @@
...
+        <dependency>
+            <groupId>commons-pool.wso2</groupId>
+            <artifactId>commons-pool</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.wso2.carbon.messaging</groupId>
+            <artifactId>org.wso2.carbon.messaging</artifactId>
+        </dependency>
+        <dependency>
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml:48]: Do we need this dependency for stat component ?

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml [hunk 3]
```diff
@@ -31,5 +32,106 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.eclipse.osgi</groupId>
+            <artifactId>org.eclipse.osgi</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.eclipse.osgi</groupId>
+            <artifactId>org.eclipse.osgi.services</artifactId>
+        </dependency>
+        <dependency>
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml:44]: I guess we can remove these dependencies.

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/Metrics.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import org.wso2.carbon.metrics.impl.MetricsLevelConfigException;
+import org.wso2.carbon.metrics.impl.MetricsLevelConfiguration;
+import org.wso2.carbon.metrics.impl.util.ConsoleReporterBuilder;
+import org.wso2.carbon.metrics.impl.util.DASReporterBuilder;
+import org.wso2.carbon.metrics.impl.util.JmxReporterBuilder;
+import org.wso2.carbon.metrics.manager.MetricManager;
+import org.wso2.carbon.metrics.manager.MetricService;
+import org.wso2.carbon.metrics.manager.ServiceReferenceHolder;
+
+/**
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/Metrics.java:15]: Need to add proper documentation.

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/internal/NettyTransportStatActivator.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.wso2.carbon.messaging.MessagingHandler;
+import org.wso2.carbon.transport.http.netty.statistics.MetricReporter;
+import org.wso2.carbon.transport.http.netty.statistics.Metrics;
+import org.wso2.carbon.transport.http.netty.statistics.StatisticsHandler;
+import org.wso2.carbon.transport.http.netty.statistics.TimerHolder;
+
+/**
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/internal/NettyTransportStatActivator.java:29]: Lets add proper documentation.

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/temp/ConnectionMetricsHolder.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/temp/ConnectionMetricsHolder.java]: I think we don't need these classes that are in the temp folder.

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Util.java [hunk 7]
```diff
@@ -26,7 +26,9 @@
 import io.netty.handler.codec.http.HttpVersion;
 import org.wso2.carbon.messaging.CarbonMessage;
 import org.wso2.carbon.messaging.Constants;
-
+//import org.wso2.carbon.transport.http.netty.latency.metrics.ConnectionMetricsHolder;
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Util.java]: Commented code is not needed.

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/disruptor/handler/CarbonDisruptorEventHandler.java [hunk 8]
```diff
@@ -34,6 +34,10 @@ public CarbonDisruptorEventHandler() {
 
     @Override
     public void onEvent(CarbonDisruptorEvent carbonDisruptorEvent, long l, boolean b) throws Exception {
+
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/disruptor/handler/CarbonDisruptorEventHandler.java]: Un-necessary spacing.

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/pool/ConnectionManager.java [hunk 9]
```diff
@@ -179,6 +179,20 @@ public TargetChannel getTargetChannel(HttpRoute httpRoute, SourceHandler sourceH
         if (targetChannel != null) {
             targetChannel.setHttpRoute(httpRoute);
             targetChannel.setCorrelatedSource(sourceHandler);
+            // Transfer the metrics data holders to the Target handler
```
Comment (shafreenAnfar) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/pool/ConnectionManager.java]: I think we can remove this code.

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml (hunk 1 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/pom.xml]: removed

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/Metrics.java (hunk 4 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/Metrics.java:15]: added

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/internal/NettyTransportStatActivator.java (hunk 5 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/internal/NettyTransportStatActivator.java:29]: addded

File: http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/temp/ConnectionMetricsHolder.java (hunk 6 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty.statistics/src/main/java/org/wso2/carbon/transport/http/netty/statistics/temp/ConnectionMetricsHolder.java]: yes, removed

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Util.java (hunk 7 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/Util.java]: removed

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/disruptor/handler/CarbonDisruptorEventHandler.java (hunk 8 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/common/disruptor/handler/CarbonDisruptorEventHandler.java]: refactored

File: http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/pool/ConnectionManager.java (hunk 9 above)
Comment (chamil321) [inline http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/pool/ConnectionManager.java]: removed

**Pull request merged**